app.py

```diff
@@ -87,5 +96,79 @@ def eli3() -> None:
         st.write(answer)
 
 
+def early_year_activity_plan() -> None:
+    """Come up with activities for children."""
+    st.title("Generating activity plans grounded in EY foundation stages")
+    auth_openai()
+    areas_of_learning_desc = read_json("src/genai/eyfs/areas_of_learning.json")
+    aol = list(areas_of_learning_desc.keys())
+
+    with st.sidebar:
+        # Select a model, temperature and number of results
+        selected_model = st.radio(label="**OpenAI model**", options=["gpt-3.5-turbo", "gpt-4"], index=1)
+        description = "<THIS IS WHERE THE GENERATOR WILL SHOW THE RESULTS>"
+        n_results = 10
+        temperature = st.slider(label="**Temperature**", min_value=0.0, max_value=1.0, value=0.6, step=0.1)
+
+    # Select the areas of learning
+    areas_of_learning = st.multiselect(label="**Areas of learning**", options=aol, default=aol)
+    areas_of_learning_text = [v for k, v in areas_of_learning_desc.items() if k in areas_of_learning]
+
+    # Describe each Area of Learning in an expanding window
+    with st.expander("**Areas of Learning Description**"):
+        for k, v in areas_of_learning_desc.items():
+            if k in areas_of_learning:
+                st.write(f"#### {k}")
+                st.write(v.split("##")[-1])
+
+    areas_of_learning_text = "\n\n".join(areas_of_learning_text)
+    location = st.selectbox(label="**Location**", options=["Indoor", "Outdoor", "Indoor or Outdoor"], index=2)
+
+    # Create the messages
+    paths = [
+        "src/genai/eyfs/prompts/system.json",
+        "src/genai/eyfs/prompts/context_and_task.json",
+        "src/genai/eyfs/prompts/constraints.json",
+        "src/genai/eyfs/prompts/situation.json",
+    ]
+
+    messages = [read_json(path) for path in paths]
+
+    # Get the user input
+    description = st.text_input(
+        label="**What's the topic you want activities for?**",
+        value="Let's create activities educating children on how whales breath",
+        help="Prompt the LLM with a some text and it will generate an activity plan for you.",
```

Comment:
Very minor, but just suggesting using less acronyms
```suggestion
        help="Prompt the large language model with a some text and it will generate an activity plan for you.",
```